src/cadenza.js

```diff
@@ -275,6 +279,72 @@ export class CadenzaClient {
     );
   }
 
+  /**
+   * Fetch data from a view.
+   *
+   * @param {WorkbookViewSource} source - The workbook view to fetch data from
+   * @param {object} options - Options
+   * @param {string} options.accept - The media type to use for the data
+   * @param {AbortSignal} [options.signal] - A signal to abort the data fetching
+   * @return {Promise<Response>} A Promise for when the data has loaded
+   * @throws For an invalid workbook view source
+   */
+  fetchData(source, { accept, signal }) {
+    this.#log('CadenzaClient#fetchData', accept);
+    const params = new URLSearchParams({
+      ...(accept && { mediaType: accept }),
+    });
+    return this.#fetch(resolvePath(source), { params, signal });
+  }
+
+  /**
+   * Download data from a view.
+   *
+   * @param {WorkbookViewSource} source - The workbook view to download data from
+   * @param {object} options - Options
+   * @param {string} options.accept - The media type to use for the data
+   * @param {string} [options.filename] - An optional filename
+   * @param {AbortSignal} [options.signal] - A signal to abort the download
+   * @throws For an invalid workbook view source
+   */
+  downloadData(source, { accept, filename, signal }) {
+    this.#log('CadenzaClient#downloadData', accept);
+    assert(validMediaType(accept), 'invalid media type');
+    download(this.fetchData(source, { accept, signal }), {
```

Comment:
I had hoped that the <a download> attribute overrides the filename in the Content-Disposition header, but unfortunately that's not the case.

Loading the data in-memory (which is what happens when creating an object URL) is not OK, I think, because these files might be big.

Proposal: Add another optional "filename" parameter to the endpoint. If present, that's used instead of the generated one.

Reply:
I had not taken into account that loading data in-memory might be an issue. Changed to make the browser handle the download.

Thanks for the proposal, it has been implemented.